addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -1344,21 +1345,14 @@ var VideoWidget = MediaWidget.extend({
         }
 
         var $content = query.$video;
-        if (!$content) {
-            switch (query.errorCode) {
-                case 0:
-                    $content = $('<div/>', {
-                        class: 'alert alert-danger o_video_dialog_iframe mb-2 mt-2',
-                        text: _t("The provided url is not valid"),
-                    });
-                    break;
-                case 1:
-                    $content = $('<div/>', {
-                        class: 'alert alert-warning o_video_dialog_iframe mb-2 mt-2',
-                        text: _t("The provided url does not reference any supported video"),
-                    });
-                    break;
-            }
+        if (!$content && (query.errorCode === 0 || query.errorCode === 1)) {
+            const type = query.errorCode === 0 ? 'danger' : 'warning';
+            this.error = query.errorCode === 0 ? _t("The provided url is not valid") :
+                _t("The provided url does not reference any supported video");
+            $content = $('<div/>', {
+                class: `alert alert-${type} o_video_dialog_iframe mb-2 mt-2`,
+                text: this.error,
+            });
```

Comment:
Sounds like "POV refactoring": you prefer this way but some prefer the switch (I do, it's easier to figure quickly), shouldn't be part of this fix IMO.
Just add `this.error = xx` in both cases, it will also reduce the whole diff to only 2 lines instead of this diff which looks like you rewritten the whole part but actually no :p

Reply:
I don't like switch when there is less than 3 cases but I understand your point anyway, I will minimize the numbers of line I will touch :+1: